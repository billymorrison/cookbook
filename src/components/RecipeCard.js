import React from "react";
import styled from "styled-components";
import {useHistory} from "react-router-dom";

const Card = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  max-height: 300px;
  border-radius: 5px;
  padding: 5px;
  margin: 40px 0 0 0;
  width: 100%;
  cursor: pointer;
  background-color: #f8a03f;
  -moz-box-shadow:    0px 1px 11px -1px rgba(148,110,0,0.55);
  -webkit-box-shadow: 0px 1px 11px -1px rgba(148,110,0,0.55);
  box-shadow:         0px 1px 11px -1px rgba(148,110,0,0.55);
`;

const CardHeader = styled.div`
  width: 100%;
  height: 90px;
  background-color: lightgray;
  display: flex;
  align-items: center;
  justify-content: space-around;
`;

const CardTitle = styled.h4`
  font-weight: bold;
  margin: 0;
  font-size: 1.9rem;
  padding: 5px 10px;
`;

const InfoBlock = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: space-evenly;
  width: 60%;
`

const Info = styled.div`
  display: flex;
  font-size: 1.1rem;
  padding: 5px 10px;
  margin: 0;

  ${({ italic }) => italic && `font-style: italic;`}
`;

const Image = styled.img`
  display: block;
  max-height:300px;
  width: auto;
  height: auto;
`

function PropertyCard({recipe}) {
  const history = useHistory();
  return (
    <Card onClick={() => history.push(`/recipe/${recipe._id}`)} data-testid="recipeCard">
      {recipe.imageUrl 
        ? <Image src={recipe.imageUrl} /> 
        : <CardHeader><i className="fas fa-hamburger"></i></CardHeader>
      }
      <InfoBlock>
        <CardTitle>{recipe.title || "Food Recipe"}</CardTitle>
        <Info italic>
          <i className="fas fa-clock fa-lg" style={{width: '30px'}}/>
          <div>
            <span> Prep: {recipe.prepTime}</span>
            <br />
            <span> Cook: {recipe.cookTime}</span>
          </div>
        </Info>
        <Info>
          <i className="fas fa-signal" style={{width: '30px'}} />
          {` ${recipe.difficulty}`}
        </Info>
        <Info>
          <i className="fas fa-at" style={{width: '30px'}} />
          {` ${recipe.userNiceName}`}
        </Info>
      </InfoBlock>
    </Card>
  );
}

export default PropertyCard;
