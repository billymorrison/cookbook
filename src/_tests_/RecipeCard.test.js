import { shallow, mount, render } from "enzyme";
import RecipeCard from "../components/RecipeCard";

it("expect to render card component", () => {
  const recipe = {
    _id: "5ff4bcbdc91e0e20e2a5be26",
    title: "Mexican chicken burger",
    displayName: "tomo",
    prepTime: "15 mins",
    cookTime: "60 mins",
    difficulty: "Hard",
    serves: 4,
    ingredients: "onion",
    method:
      "In a large, heavy saucepan, heat the oil. Add the onion, garlic, green chilli, ginger and some seasoning. Fry on a medium heat for 10 mins or until soft",
    __v: 0,
  };
  expect(shallow(<RecipeCard recipe={recipe} />).length).toEqual(1);
  expect(toJson(<RecipeCard />)).toMatchSnapshot();
});
