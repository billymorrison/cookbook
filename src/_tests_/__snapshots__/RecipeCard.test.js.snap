// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="sc-bdfBwQ iVwPzy"
  data-testid="recipeCard"
  onClick={[Function]}
>
  <div
    className="sc-gsTCUz eUXZjv"
  >
    <i
      className="fas fa-hamburger"
    />
  </div>
  <div
    className="sc-hKgILt bKKnog"
  >
    <h4
      className="sc-dlfnbm hhHbbO"
    >
      Mexican chicken burger
    </h4>
    <div
      className="sc-eCssSg kvMIzk"
    >
      <i
        className="fas fa-clock fa-lg"
        style={
          Object {
            "width": "30px",
          }
        }
      />
      <div>
        <span>
           Prep: 
          15 mins
        </span>
        <br />
        <span>
           Cook: 
          60 mins
        </span>
      </div>
    </div>
    <div
      className="sc-eCssSg bJUjBG"
    >
      <i
        className="fas fa-signal"
        style={
          Object {
            "width": "30px",
          }
        }
      />
       Hard
    </div>
    <div
      className="sc-eCssSg bJUjBG"
    >
      <i
        className="fas fa-at"
        style={
          Object {
            "width": "30px",
          }
        }
      />
       undefined
    </div>
  </div>
</div>
`;
